[DOCS-3339] Add Observability Pipelines Docs (#14245)
DataDog/documentation: content/en/integrations/observability_pipelines/working_with_data.md

    tag: Documentation
    text: Parsing metadata emitted by AWS EC2 instance
  - link: https://vector.dev/docs/reference/configuration/transforms/lua/
    tag: Documentation
    text: Modifying events with Lua
  - link: https://vector.dev/docs/reference/configuration/transforms/tag_cardinality_limit/ 
    tag: Documentation
    text: Limit the cardinality of tags on metrics to safeguard against cardinality explosion
  - link: https://vector.dev/docs/reference/configuration/transforms/log_to_metric/
    tag: Documentation
    text: Convert logs to metric events
  - link: https://vector.dev/docs/reference/configuration/transforms/metric_to_log/ 
    tag: Documentation
    text: Convert metrics to log events
  - link: https://vector.dev/docs/reference/configuration/transforms/geoip/
    tag: Documentation
    text: Enrich events with GeoIP metadata
  - link: https://docs.datadoghq.com/agent/vector_aggregation/ 
    tag: Documentation
    text: Configure Datadog agents to send data to Vector aggregators
  - link: /integrations/observability_pipelines/vector_configurations/
    tag: Documentation
    text: Learn more about Vector configurations
---

## Overview

> Vector enables you to shape and transform observability data. Similar to Logging without Limits™ pipelines, you can configure Vector pipelines that are composed of a series of Vector `transform` components. These transforms allow you to parse, structure, and  enrich data with built-in type safety.

Author: ```suggestion
Vector enables you to shape and transform observability data. Similar to Logging without Limits™ pipelines, you can configure Vector pipelines that are composed of a series of Vector `transform` components. These transforms allow you to parse, structure, and enrich data with built-in type safety.
```

- [Convert syslog values][8] to read-able values.
- Enrich values by using [enrichment tables][22].
- [Manipulate][9] IP values.
- [Parse][10] values with custom rules (for example, grok, regex, and so on) and out-of-the-box functions (for example, syslog, apache, VPC flow logs, and so on).
- Manipulate event [metadata][11] and [paths][12].

...and [more][1]

To get started, see the following example for a basic remap transform that contains a VRL program in the `source` field:

{{< tabs >}}
{{% tab "YAML" %}}

```yaml
transforms:
  modify:
    type: remap
    inputs:
      - previous_component_id
    source: |2
        del(.user_info)
        .timestamp = now()
```

{{% /tab %}}
{{% tab "TOML" %}}

```toml
[transforms.modify]
type = "remap"
inputs = ["previous_component_id"]
source = '''
  del(.user_info)
  .timestamp = now()
'''
```

{{% /tab %}}
{{% tab "JSON" %}}

```json
{
  "transforms": {
    "modify": {
      "type": "remap",
      "inputs": [
        "previous_component_id"
      ],
      "source": "  del(.user_info)\n  .timestamp = now()\n"
    }
  }
}
```

{{% /tab %}}
{{< /tabs >}}

> In this example, the `type` field is set to a `remap` transform. The `inputs` field defines where it will receive events from the previously defined `previous_component_id` source. The first line in the `source` field deletes the `.user_info` field. At scale, dropping fields is particularly useful for reducing the payload of your events and cutting down on spend for your downstream services. 

Author: ```suggestion
In this example, the `type` field is set to a `remap` transform. The `inputs` field defines where it receives events from the previously defined `previous_component_id` source. The first line in the `source` field deletes the `.user_info` field. At scale, dropping fields is particularly useful for reducing the payload of your events and cutting down on spend for your downstream services. 
```

      "type": "remap",
      "inputs": [
        "previous_component_id"
      ],
      "source": "   . = parse_json!(string!(.message))\n   .timestamp = to_unix_timestamp(to_timestamp!(.timestamp))\n   del(.username)\n   .message = downcase(string!(.message))\n"
    }
  }
}
```

{{% /tab %}}
{{< /tabs >}}

This configuration returns the following:

```
{
  "message": "success",
  "status": 200,
  "timestamp": 1614626364
}
```

## Sample, reduce, filter, and aggregate data

Sampling, reducing, filtering, and aggregating are common transforms to reduce the volume of observability data delivered to downstream services. Vector offers a variety of ways to control your data volume:

- [Sample events][15] based on supplied criteria and at a configurable rate.
- [Reduce and collapse][16] multiple events into a single event.
- Remove unnecessary fields.
- [Deduplicate][17] events. 
- [Filter events][18] based on a set of conditions.
- [Aggregate multiple metric events][19] into a single metric event based on a defined interval window.
- Convert [metrics to logs][20].

**TBD** For a detailed guide on how to use these transform for cost control, see the [cost control guide].

## Route data

> Another commonly used transform is `route`, which allows you to split a stream of events into multiple substreams based on  supplied conditions. This is useful when you need to send observability data to different destinations or operate differently on streams of data based on their use case. 

Author: ```suggestion
Another commonly used transform is `route`, which allows you to split a stream of events into multiple substreams based on supplied conditions. This is useful when you need to send observability data to different destinations or operate differently on streams of data based on their use case. 
```
